frontend/src/scenes/dashboard/DashboardTemplateEditorLogic.ts

```diff
@@ -0,0 +1,116 @@
+import { editor } from 'monaco-editor'
+import { lemonToast } from '@posthog/lemon-ui'
+import { actions, connect, kea, listeners, path, reducers } from 'kea'
+import { loaders } from 'kea-loaders'
+import api from 'lib/api'
+import { DashboardTemplateType } from '~/types'
+import { dashboardTemplatesLogic } from './dashboards/templates/dashboardTemplatesLogic'
+
+import type { dashboardTemplateEditorLogicType } from './DashboardTemplateEditorLogicType'
+
+export const dashboardTemplateEditorLogic = kea<dashboardTemplateEditorLogicType>([
+    path(['scenes', 'dashboard', 'NewDashboardTemplateLogic']),
+    connect(dashboardTemplatesLogic),
+    actions({
+        setDashboardTemplateJSON: (dashboardTemplateJSON: string) => ({ dashboardTemplateJSON }),
+        setOpenNewDashboardTemplateModal: (openNewDashboardTemplateModal: boolean) => ({
+            openNewDashboardTemplateModal,
+        }),
+        createDashboardTemplate: (dashboardTemplateJSON: string) => ({ dashboardTemplateJSON }),
+        setDashboardTemplateId: (id: string | undefined) => ({ id }),
+        closeNewDashboardTemplateModal: true,
+        updateValidationErrors: (markers: editor.IMarker[] | undefined) => ({ markers }),
+    }),
+    reducers({
+        dashboardTemplateJSON: [
+            '' as string,
+            {
+                setDashboardTemplateJSON: (_, { dashboardTemplateJSON }) => dashboardTemplateJSON,
+            },
+        ],
+        validationError: [
+            '' as string,
+            {
+                updateValidationErrors: (_, { markers }) => {
+                    if (!markers || markers.length === 0) {
+                        console.log('returning undefined')
+                        return ''
+                    } else {
+                        console.log('updated with markers', markers)
+                        return markers.map((marker) => marker.message).join(', ')
+                    }
+                },
+            },
+        ],
+        isOpenNewDashboardTemplateModal: [
+            false as boolean,
+            {
+                setOpenNewDashboardTemplateModal: (_, { openNewDashboardTemplateModal }) =>
+                    openNewDashboardTemplateModal,
+                closeNewDashboardTemplateModal: () => false,
+            },
+        ],
+        id: [
+            undefined as string | undefined,
+            {
+                setDashboardTemplateId: (_, { id }) => id,
+            },
+        ],
+    }),
+    loaders(({ values }) => ({
+        dashboardTemplate: [
+            null as DashboardTemplateType | null,
+            {
+                createDashboardTemplate: async () => {
+                    const response = await api.create(
+                        '/api/projects/@current/dashboard_templates',
```

Comment:
We should move these API interactions into `api.ts` completely, which lets us use specific team over `@current`